src/DIRAC/Core/Security/IAMService.py

```diff
@@ -0,0 +1,123 @@
+        result = gConfig.getOptionsDict(f"/Resources/IdProviders/{id_provider}")
+        if result["OK"]:
+            self.iam_url = result["Value"]["issuer"]
+            gLogger.verbose("Using IAM server", self.iam_url)
+        else:
+            raise ValueError(f"/Resources/IdProviders/{id_provider}")
+
+        self.userDict = None
+        self.access_token = access_token
+
+    def _getIamUserDump(self):
+        """List the users from IAM"""
+
+        headers = {"Authorization": f"Bearer {self.access_token}"}
+        iam_list_url = f"{self.iam_url}/scim/Users"
+        iam_users = []
+        startIndex = 1
+        totalResults = 1000  # total number of users
+        itemsPerPage = 10
+        while startIndex <= totalResults:
+            resp = requests.get(iam_list_url, headers=headers, params={"startIndex": startIndex})
+            resp.raise_for_status()
+            data = resp.json()
+            # These 2 should never change, but just to be sure...
+            totalResults = data["totalResults"]
+            itemsPerPage = data["itemsPerPage"]
+
+            startIndex += itemsPerPage
+            iam_users.extend(data["Resources"])
+        return iam_users
+
+    @staticmethod
+    def convert_iam_to_voms(iam_output):
+        """Convert an IAM entry into the voms style, i.e. DN based"""
+        converted_output = {}
+
+        for cert in iam_output["urn:indigo-dc:scim:schemas:IndigoUser"]["certificates"]:
+            cert_dict = {}
+            dn = convert_dn(cert["subjectDn"])
+            ca = convert_dn(cert["issuerDn"])
+
+            cert_dict["CA"] = ca
+            cert_dict["nickname"] = iam_output["userName"]
+            # This is not correct, we take the overall status instead of the certificate one
+            # however there are no known case of cert suspended while the user isn't
+            cert_dict["certSuspended"] = not iam_output["active"]
+            # There are still bugs in IAM regarding the active status vs voms suspended
+
+            cert_dict["suspended"] = not iam_output["active"]
+            # The mail may be different, in particular for robot accounts
+            cert_dict["mail"] = iam_output["emails"][0]["value"].lower()
+
+            # https://github.com/indigo-iam/voms-importer/blob/main/vomsimporter.py
+            roles = []
+
+            for role in iam_output["groups"]:
+                role_name = role["display"]
+                if "/" in role_name:
+                    role_name = role_name.replace("/", "/Role=")
+                roles.append(f"/{role_name}")
+
+            cert_dict["Roles"] = roles
+            converted_output[dn] = cert_dict
+        return converted_output
+
+    def getUsers(self):
+        self.iam_users_raw = self._getIamUserDump()
+        users = {}
+        errors = 0
+        for user in self.iam_users_raw:
+            try:
+                users.update(self.convert_iam_to_voms(user))
```

Comment:
So IAM returns a list of dictionaries where each dictionary has one key (the username)?

Might be nice to put some example responses in docstrings to make it easier to follow.